### PR TITLE
Create Softmax layer in activation decomposition instead of Activation layer

### PR DESCRIPTION
**Pull request by reuvenperetz**
## Pull Request Description:

In MP we need to know the axis used in Softmax for KL-Divergence computation. In activation decomposition, we created a simple 'Activation' layer thus we can not use the axis info when computing MP distance. This commit changes the decomposition to create a Softmax layer, thus enabling the KL-Divergence in MP distance computation.

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/activation_decomposition.py [hunk 1]
```diff
@@ -70,18 +70,30 @@ def substitute(self,
...
-            ACTIVATION: op2d_node.framework_attr.get(ACTIVATION)}
-
-        activation_node = common.graph.BaseNode(activation_node_name,
-                                                activation_fw_attr,
-                                                op2d_node.output_shape,
-                                                op2d_node.output_shape,
-                                                {},
-                                                Activation)
+        # Softmax is a special case where we need to know the default axis parameter used
+        # (for the distance metric in MP for KL-Divergence), and for this reason we create
```
Comment (ofirgo) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/activation_decomposition.py:74]: I would remove the cause reference (MP and KL) from here, it can be confusing if you just looked at this file without context.

**Review by ofirgo — APPROVED** [review]
